### DISCUSSION TITLE
How does trim_nonpercolating_paths work?

### DISCUSSION THREAD

**Discussion by ma-sadeghi** [PMEAL/porespy#352] [Q&A] [answered]
Here's a snippet, creating blobs in 2D, and then trimming non-percolating paths along the y-axis. I don't quite get why we still see some floating regions.

```python
import porespy as ps
im = ps.generators.blobs(shape=[50, 50], porosity=0.4)
im = ps.filters.trim_nonpercolating_paths(im, inlet_axis=1, outlet_axis=1)
```
![image](https://user-images.githubusercontent.com/14086031/106209618-563f2e00-6193-11eb-9159-acba983472cb.png)

**Comment by jgostick** [comment, marked as answer]
that looks like a bug to me.  I've actually never like this axis business....I prefer passing in boolean masks of inlets and outlets.  I updated this function to also accepts those, but I have just confirmed that it is also broken.

**Reply by jgostick** [reply]
This bug has been fixed and is now part of the `dev` branch.  It will be merged into an official release soon
